### PR TITLE
Bayesian Optimization integration tests

### PR DESCRIPTION
**Pull request by clinton0313**
Changes are below. I was unable to run the final two Efficient Global Optimization tests in the test_bayesian_optimization for scaled_branin for both 
dropout and dropconnect. The process kept killing my computer..Maybe you could give it a run?

- Implemented BO tests
- Added architecture test to check for scalars
- Changed plotting function for plotly to show sequence of predictions.
- Removed separate dropout test files and merged to original test files. Marked mcdropout tests. Refactored some tests.
- Quick and dirty fix to reset learning rates in MCDropout optimize - can be superceded.
- Moved BO tests to original file. Updated layers. Updated MCDropout notebook.
- Removed separate bayesian optimization pytest file.

### CODE REVIEW

**Comment by ampudia19** [comment]
I can definitely run it tomorrow, today my laptop is already going to suffer enough as is with some Ruben tasks.

File: trieste/models/keras/architectures.py [hunk 1]
```diff
@@ -324,6 +324,9 @@ def _gen_output_layer(self) -> tf.keras.Model:
 
     def call(self, inputs: tf.Tensor) -> tf.Tensor:
 
+        if inputs.shape.rank == 1:
+            inputs = tf.expand_dims(inputs, axis=-1)
+
```
Comment (ampudia19) [inline trieste/models/keras/architectures.py:330]: are we not sure we would rather catch this as an error, rather than be lenient and just expand it internally?

File: trieste/models/keras/models.py [hunk 2]
```diff
@@ -351,17 +351,19 @@ def __init__(
...
-                        monitor="loss", factor=0.3, patience=15
+                        monitor="loss", factor=0.3, patience=20
                     )
                 ],
             }
 
         if self.optimizer.loss is None:
             self.optimizer.loss = "mse"
 
+        self._learning_rate = self.optimizer.optimizer.learning_rate.numpy()
```
Comment (ampudia19) [inline trieste/models/keras/models.py:365]: We had a very similar solution, so feel free to prioritize this one. Results will be the same.
